syntax = "proto3";

package User;

service UserService {
    rpc AddUser(UserRequest) returns (UserResponse);
    rpc GetUser(UserIdRequest) returns (UserResponse);
    rpc UpdateUser(UserRequest) returns (UserResponse);
    rpc DeleteUser(UserIdRequest) returns (UserResponse);
}

//adicionado campo de idade
message UserRequest {
    int32 id = 1;
    string name = 2;
    string email = 3;
    int32 age = 4;
}

message UserIdRequest {
    int32 id = 1;
}

message UserResponse {
    int32 id = 1;
    string name = 2;
    string email = 3;
    int32 age = 4;
    string msg = 5;
}
